t/34-git.t

```diff
@@ -58,43 +58,43 @@ my $case_dir_ok = "file://$git_dir#$head";
 my $case_dir = "file://$git_dir#abcdef";
 
 subtest 'failing clone' => sub {
-    %bmwqemu::vars = (
-        CASEDIR => $case_dir,
-    );
+    %bmwqemu::vars = (CASEDIR => $case_dir);
     my $path;
     my $out = combined_from {
         eval { $path = checkout_git_repo_and_branch('CASEDIR', retry_count => 0) };
     };
     my $error = $@;
     like $error, qr{Could not find 'abcdef' in complete history in cloned Git repository '\Q$case_dir\E'}, "Error message when trying to clone wrong git hash";
-    like $out, qr{Cloning git URL.*Fetching more remote objects.*Enumerating objects}s, 'git fetch was called to get more commits';
+    like $out, qr{Fetching 'abcdef' from origin manually}s, 'manual git fetch for revspec was attempted';
+    like $out, qr{Cloning git URL.*Fetching more remote objects.*Enumerating objects}s, 'git fetch with --depth option was attempted';
 };
 
-cleanup();
-
 subtest 'successful clone' => sub {
     my $path;
-    %bmwqemu::vars = (
-        CASEDIR => $case_dir_ok,
-    );
-    my $out = combined_from {
-        $path = checkout_git_repo_and_branch('CASEDIR');
-    };
-    is $path, $clone_dir, 'checkout_git_repo_and_branch returned correct path';
-    like $out, qr{Cloning git URL.*Fetching more remote objects}s, 'git clone was called again to fetch a git hash';
 
-    %bmwqemu::vars = (
-        CASEDIR => $case_dir_ok,
-    );
-    $out = combined_from {
-        $path = checkout_git_repo_and_branch('CASEDIR');
+    subtest 'fetch commit manually but directly' => sub {
+        cleanup();
+        %bmwqemu::vars = (CASEDIR => $case_dir_ok);
+        my $out = combined_from { $path = checkout_git_repo_and_branch('CASEDIR') };
+        is $path, $clone_dir, 'checkout_git_repo_and_branch returned correct path';
+        like $out, qr{Cloning git URL.*Fetching '.*' from origin manually}s, 'git clone and fetch were called again to fetch rev manually';
+        unlike $out, qr{Cloning git URL.*Fetching more remote objects}s, 'no need to resort to general fetch';
     };
-    is $path, $clone_dir, 'checkout_git_repo_and_branch with existing local directory returned correct path';
-    like $out, qr{Skipping to clone.*tmpgitrepo already exists}, 'Log says that local directory already exists';
-
-    eval {
-        bmwqemu::save_vars(no_secret => 1);
+    subtest 'skip cloning when repo already exists (not even switching to the correct branch)' => sub {
+        %bmwqemu::vars = (CASEDIR => $case_dir_ok);
+        my $out = combined_from { $path = checkout_git_repo_and_branch('CASEDIR') };
+        is $path, $clone_dir, 'checkout_git_repo_and_branch with existing local directory returned correct path';
+        like $out, qr{Skipping to clone.*tmpgitrepo already exists}, 'Log says that local directory already exists';
+    };
+    subtest 'fetch commit manually by fetching repeatedly with increasing depth' => sub {
+        cleanup();
+        %bmwqemu::vars = (CASEDIR => $case_dir_ok);
```

Comment:
The tests seem to target the changes you made. But I wonder if the case you mentioned `doesn't handle the case when the branch is actually just a ref like refs/pull/<pr_number>/merge` shouldn't be tested explicitly as well

Reply:
A ref is a ref. So this shouldn't be a special case from our side. Of course we rely on GitHub actually creating those special refs but it should be good enough to test GitHub's behavior on the level of the example distribution.